CDCgov/prime-reportstream
# 13505: Replace e2e up test with more performant version

This PR removes the old Universal Pipeline end2end test and replaces it with a more performant test. The goal of this new test is not to verify each step in the report lifecycle or lineage, but to verify that it is sent to the correct receivers and that the contents match what we expect them to be.

**Note: Still new here so please feel free to correct any wording issues in documentation/logging/variable naming that may stem from incomplete understanding.**

Test Steps:
1. Run `./gradlew testEnd2EndUP`
2. Expect to see two reports created. One for the FULL_ELR workflow and the other for ELIMS.
3. Verify the receivers and sent file names in the output via `curl --location 'http://localhost:7071/api/waters/report/[report_id]/history/'`
4. Running in staging requires setting the AzureWebJobsStorage environment variable. Because that variable contains semicolons it will need to be set in CLI like `export AzureWebJobsStorage="<key from azure>"`

## Changes
- Replaced the old UP e2e test with a test that posts a report. It then verifies the sent reports were sent to the correct receivers. It also pulls the resulting files from the blob store and verifies that the contents are what we expect.
- This allowed a reduction in local test run time from about 8 minutes to under 1.5 minutes.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #13505

## TODO
- https://app.zenhub.com/workspaces/platform-6182b02547c1130010f459db/issues/gh/cdcgov/prime-reportstream/14336
